### PR TITLE
Add cluster upgrade tests (1.19.6-1.20.3) for CAPA

### PR DESCRIPTION
**Pull request by rayandas**
This PR adds a periodics for cluster upgrade test for CAPA.

Signed-off-by: Rayan Das <rayandas91@gmail.com>

Fixes: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/3087

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-periodics.yaml [hunk 1]
```diff
@@ -40,6 +40,50 @@ periodics:
...
+    preset-aws-ssh: "true"
+    preset-aws-credential: "true"
+  extra_refs:
+  - org: kubernetes-sigs
+    repo: cluster-api-provider-aws
+    base_ref: main
+    path_alias: "sigs.k8s.io/cluster-api-provider-aws"
+  spec:
+    containers:
+    - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220214-0aa8fe1d3a-master
```
Comment (sedefsavas) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-periodics.yaml:61]:
```suggestion
    - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20211217-ea95cec1d4-1.23
```
Let's not use the master kubekins image for these tests to reduce risk of unrelated failures.

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-periodics.yaml [hunk 2]
```diff
@@ -40,6 +40,50 @@ periodics:
     testgrid-tab-name: periodic-e2e-main
     testgrid-alert-email: sig-cluster-lifecycle-cluster-api-alerts@kubernetes.io
     testgrid-num-failures-to-alert: "2"
+- name: periodic-cluster-api-provider-aws-e2e-upgrade-1-19-1-20-main
+  decorate: true
+  decoration_config:
+    timeout: 5h
+  interval: 12h
```
Comment (sedefsavas) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-periodics.yaml:47]:
```suggestion
  interval: 24h
```

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-periodics.yaml (hunk 1 above)
Comment (rayandas) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-periodics.yaml:61]: Yeah right. Updating it.

**Comment by rayandas** [comment]
Updated the PR with suggested changes.

**Pull request closed without merging**